client-side/js/options/colorPallet.js

```diff
@@ -0,0 +1,100 @@
+window.options.colorPallet = {
+  name: 'color pallet',
+  run: function (e) {
+    const element = document.createElement('div')
+    element.innerHTML = `
+      <style>
+        .pallet {
+          display: block;
+          grid-template-columns: repeat(1, 1fr);
+          gap: 5px;
+        }
+        input {
+          width: 25px;
+          height: 25px;
+          margin: 5px 0 0 5px;
+          border: 2px solid black;
+          border-radius: 50%;
+        }
+        input:hover {
+          border-color: grey;
+        }
+        #white {background-color: white;}
+        #black {background-color: black;}
+        #red {background-color: rgb(255, 0, 0);}
+        #orange {background-color: rgb(255, 111, 0);}
+        #yellow {background-color: rgb(255, 247, 0);}
+        #green {background-color: rgb(53, 189, 22);}
+        #cyan {background-color: rgb(0, 255, 212);}
+        #blue {background-color: rgb(0, 145, 255);}
+        #purple {background-color: rgb(187, 0, 255);}
+        #pink {background-color: rgb(255, 89, 177);}
```

Comment:
something to note about the CSS we create in our options is that these get added "globally" to the app, which means that creating these ids here means no other options (or any part of our app) can use these id names, which in theory is ok.. but b/c the names are so generic (*white, black, red, etc*), there's a possibility that someone else could accidentally also create elements w/these id names.

there are a few things we could do to avoid this potential future conflict:

1. there are ways to create "[custom elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements)" which allow u to encapsulate the styles so that they only apply to ur custom element && don't effect any other parts of ur page/app using something called a [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM)... these are parts of the [Web Components API](https://developer.mozilla.org/en-US/docs/Web/Web_Components) (which is essentially the native Web API for creating "components", like those created using libraries like React.js && Vue.js), but that's probably a bit too much to get into for this project.

2. u could use more specific CSS query selectors, for example istead of using `#white` we could say `input[type="button"]:nth-child(1)` which would target the same element, by saying *get the input element with a type of 'button' that is the first of it's parent's children*... this would allow us to get rid of the id's on those elements.

3. but there's another approach we could take that would solve this problem && drastically reduce the amount of lines of code u have in this module... see my next comment below...